proxy/Main.cc

```diff
@@ -346,6 +346,54 @@ class DiagsLogContinuation : public Continuation
   }
 };
 
+class MemoryLimit : public Continuation
+{
+public:
+  MemoryLimit() : Continuation(new_ProxyMutex()), _memory_limit(0) { SET_HANDLER(&MemoryLimit::periodic); }
+  ~MemoryLimit() { mutex = NULL; }
+  int
+  periodic(int event, Event *e)
+  {
+    if (event == EVENT_IMMEDIATE) {
+      // rescheduled from periodic to immediate event
+      // this is the indication to terminate
+      delete this;
+      return EVENT_DONE;
+    }
+    if (_memory_limit == 0) {
+      // first time it has been run
+      _memory_limit = REC_ConfigReadInteger("proxy.config.memory.max_usage");
```

Comment:
I don't have an event scheduled if it is not configured.  I thought it would be nice to not have an event scheduled every 1 second if someone doesn't have it configured.